ReFrame breaks if user's `.bashrc` contains a `cd` to directory

Originally reported by @aurianer some time ago, but I have now verified it:

```
karakasv@tresa reframe [master] $ ./test_reframe.py --maxfail=1
============================================================================================= test session starts ==============================================================================================
platform darwin -- Python 3.7.3, pytest-4.3.0, py-1.8.0, pluggy-0.9.0
rootdir: /Users/karakasv/Repositories/reframe, inifile:
collected 621 items

unittests/test_argparser.py ..                                                                                                                                                                           [  0%]
unittests/test_buildsystems.py ...........                                                                                                                                                               [  2%]
unittests/test_check_filters.py ........                                                                                                                                                                 [  3%]
unittests/test_cli.py ....sF

=================================================================================================== FAILURES ===================================================================================================
_______________________________________________________________________________________ TestFrontend.test_check_success ________________________________________________________________________________________

self = <unittests.test_cli.TestFrontend testMethod=test_check_success>

    def test_check_success(self):
        self.more_options = ['--save-log-files']
        returncode, stdout, _ = self._run_reframe()
>       self.assertIn('PASSED', stdout)
E       AssertionError: 'PASSED' not found in "Command line: ./bin/reframe --prefix unittests/tmpnqwnoz85 --nocolor --system generic:login -C unittests/resources/settings.py -c unittests/resources/checks/hellocheck.py -p builtin-gcc --force-local -r --ignore-check-conflicts --perflogdir .rfm-perflogs --save-log-files\nReframe version: 2.20-dev2\nLaunched by user: karakasv\nLaunched on host: tresa.local\nReframe paths\n=============\n    Check prefix      : \n    Check search path : 'unittests/resources/checks/hellocheck.py'\n    Stage dir prefix     : /Users/karakasv/Repositories/reframe/unittests/tmpnqwnoz85/stage/\n    Output dir prefix    : /Users/karakasv/Repositories/reframe/unittests/tmpnqwnoz85/output/\n    Perf. logging prefix : /Users/karakasv/Repositories/reframe/.rfm-perflogs\n[==========] Running 1 check(s)\n[==========] Started on Tue Oct 22 00:00:02 2019 \n\n[----------] started processing hellocheck (C Hello World test)\n[ RUN      ] hellocheck on generic:login using builtin-gcc\n[     FAIL ] hellocheck on generic:login using builtin-gcc\n[----------] finished processing hellocheck (C Hello World test)\n\n[  FAILED  ] Ran 1 test case(s) from 1 check(s) (1 failure(s))\n[==========] Finished on Tue Oct 22 00:00:03 2019 \n\n==============================================================================\nSUMMARY OF FAILURES\n------------------------------------------------------------------------------\nFAILURE INFO for hellocheck \n  * System partition: generic:login\n  * Environment: builtin-gcc\n  * Stage directory: /Users/karakasv/Repositories/reframe/unittests/tmpnqwnoz85/stage/generic/login/builtin-gcc/hellocheck\n  * Job type: local (id=None)\n  * Maintainers: ['VK']\n  * Failing phase: compile_wait\n  * Reason: build error: standard error can be found in `rfm_hellocheck_build.err', standard output can be found in `rfm_hellocheck_build.out'\n------------------------------------------------------------------------------\n"

unittests/test_cli.py:136: AssertionError
--------------------------------------------------------------------------------------------- Captured stdout call ---------------------------------------------------------------------------------------------
./bin/reframe --prefix unittests/tmpnqwnoz85 --nocolor --system generic:login -C unittests/resources/settings.py -c unittests/resources/checks/hellocheck.py -p builtin-gcc --force-local -r --ignore-check-conflicts --perflogdir .rfm-perflogs --save-log-files
================================================================================ 1 failed, 25 passed, 1 skipped in 2.40 seconds ================================================================================
```

--- a/reframe/core/pipeline.py
+++ b/reframe/core/pipeline.py
@@ -1057,7 +1057,7 @@ class RegressionTest(metaclass=RegressionTestMeta):
         with os_ext.change_dir(self._stagedir):
             try:
                 self._build_job.prepare(build_commands, environs,
-                                        login=True, trap_errors=True)
+                                        trap_errors=True)
             except OSError as e:
                 raise PipelineError('failed to prepare build job') from e
 
@@ -1136,7 +1136,7 @@ class RegressionTest(metaclass=RegressionTestMeta):
 
         with os_ext.change_dir(self._stagedir):
             try:
-                self._job.prepare(commands, environs, login=True)
+                self._job.prepare(commands, environs)
             except OSError as e:
                 raise PipelineError('failed to prepare job') from e
 

